### PR TITLE
podutils: Add scheduling options do decoration config

### PR DESCRIPTION
**Pull request by Ressetkk**
/kind feature

This feature adds ability to define some affinities and tolerations that will be added during Pod generation. This is helpful for workloads where you want to run ProwJobs on an isolated node pool without changing the affinity for other workloads unrelated to ProwJobs

### CODE REVIEW

File: prow/apis/prowjobs/v1/types.go [hunk 1]
```diff
@@ -603,6 +618,33 @@ func (g *CensoringOptions) ApplyDefault(def *CensoringOptions) *CensoringOptions
 	return &merged
 }
 
+// ApplyDefault applies the defaults for CensoringOptions decorations. If a field has a zero value,
```
Comment (petr-muller) [inline prow/apis/prowjobs/v1/types.go:621]:
```suggestion
// ApplyDefault applies the defaults for SchedulingOptions decorations. If a field has a zero value,
```

**Review by petr-muller — APPROVED** [review]
Minor nit, otherwise code change looks good. I am not entirely convinced this is the best way to do this (not too fond of inheriting complex api elements into Prowjob) but in the absence of alternative...

My approver power is not enough anyway, this needs someone from test-infra-oncall

**Comment by petr-muller** [comment]
```
Error from server: etcdserver: request is too large
```
infra issue

/test pull-test-infra-integration

**Comment by petr-muller** [comment]
or actually... maybe we're hitting some CRD size limit? :thinking:

**Comment by Ressetkk** [comment]
> or actually... maybe we're hitting some CRD size limit? 🤔

Looks like it 🙄

**Comment by petr-muller** [comment]
```console
❯ git checkout upstream/master
❯ du --bytes config/prow/cluster/prowjob-crd/prowjob_customresourcedefinition.yaml
3797217	config/prow/cluster/prowjob-crd/prowjob_customresourcedefinition.yaml

❯ gh pr checkout 30286
❯ du --bytes config/prow/cluster/prowjob-crd/prowjob_customresourcedefinition.yaml
3864116	config/prow/cluster/prowjob-crd/prowjob_customresourcedefinition.yaml
```

So prowjob YAML is already 3.7M (how did we not fail sooner?) and this PR adds 65k more, which goes over some limit (not sure about what the limit actually is, I thought etcd limit is lower than that (around 1.5M I vaguelly recall?)

xref: https://github.com/kubernetes/kubernetes/issues/82292

**Comment by Ressetkk** [comment]
@upodroid I need to sync on that given there is a problem with etcd not accepting ProwJob CRD because it's too big.

**Comment by jihoon-seo** [comment]
Hello @Ressetkk, the Prow source code has been moved to https://github.com/kubernetes-sigs/prow on April 9, 2024.
Please consider migrate this PR (by opening a new one in there).
Thanks!

**Comment by Ressetkk** [comment]
> Hello @Ressetkk, the Prow source code has been moved to https://github.com/kubernetes-sigs/prow on April 9, 2024. Please consider migrate this PR (by opening a new one in there). Thanks!

Thanks for the heads up. I'll definitely get back to this topic soon.

**Pull request closed without merging**